Sources/Abstract/Free/Multiset+FreeCommutativeMonoid.swift

```diff
@@ -0,0 +1,167 @@
+#if SWIFT_PACKAGE
+	import Operadics
+#endif
+
+public struct Multiset<A>: Equatable where A: Hashable {
+	private var storage: [A : Int] = [:]
+	public private(set) var count = 0
+
+	public init() {}
+
+	public init <S> (_ sequence: S) where S: Sequence, S.Iterator.Element == A {
+		self.init()
+		for item in sequence {
+			self.add(item)
+		}
+	}
+
+	public mutating func add(_ item: A) {
+		storage[item, default: 0] += 1
+		count += 1
+	}
+
+	public mutating func remove(_ item: A) {
+		if let currentCount = storage[item] {
+			if currentCount > 1 {
+				storage[item, default: 2] -= 1
+			} else {
+				storage[item] = nil
+			}
+			count -= 1
+		}
+	}
+
+	public mutating func removeAll(_ item: A) {
+		while self.contains(item) {
+			remove(item)
+		}
+	}
+
+	public func count(of item: A) -> Int {
+		return storage[item] ?? 0
+	}
+
+	public func contains(_ item: A) -> Bool {
+		return count(of: item) > 0
+	}
+
+	public var anyItem: A? {
+		return storage.keys.first
+	}
+
+	public var allItems: [A] {
+		return Array(storage)
+			.flatMap {
+				Array.init(repeating: $0.key, count: $0.value)
+		}
+	}
+
+	public struct Iterator: IteratorProtocol {
+		private var multiSet: Multiset
+		public init(_ multiSet: Multiset) {
+			self.multiSet = multiSet
+		}
+
+		public mutating func next() -> (A, Int)? {
+			guard let anyItem = multiSet.anyItem else { return nil }
+            let count = multiSet.count(of: anyItem)
```

Comment:
Nit:

Reply:
@CodaFi not sure what happened here. Fixed.